core/domain/exp_domain_test.py

```diff
@@ -12737,65 +12748,69 @@ def test_all_html_strings_are_collected(self) -> None:
+        choices_subtitled_dict: List[state_domain.SubtitledHtmlDict] = [
+            {
+                'content_id': 'ca_choices_0',
+                'html': '<p>This is value1 for MultipleChoice</p>'
+            },
+            {
+                'content_id': 'ca_choices_1',
+                'html': '<p>This is value2 for MultipleChoice</p>'
+            }
+        ]
         customization_args_dict2: Dict[
-            str, Dict[str, Union[List[Dict[str, str]], bool]]
+            str, Dict[str, Union[List[state_domain.SubtitledHtmlDict], bool]]
         ] = {
-            'choices': {'value': [
-                {
-                    'content_id': 'ca_choices_0',
-                    'html': '<p>This is value1 for MultipleChoice</p>'
-                },
-                {
-                    'content_id': 'ca_choices_1',
-                    'html': '<p>This is value2 for MultipleChoice</p>'
-                }
-            ]},
+            'choices': {'value': choices_subtitled_dict},
             'showChoicesInShuffledOrder': {'value': True}
         }
+        choices_subtitled_dict = [
+            {
+                'content_id': 'ca_choices_0',
+                'html': '<p>This is value1 for ItemSelection</p>'
+            },
+            {
+                'content_id': 'ca_choices_1',
+                'html': '<p>This is value2 for ItemSelection</p>'
+            },
+            {
+                'content_id': 'ca_choices_2',
+                'html': '<p>This is value3 for ItemSelection</p>'
+            }
+        ]
         customization_args_dict3: Dict[
-            str, Dict[str, Union[List[Dict[str, str]], int]]
+            str, Dict[str, Union[List[state_domain.SubtitledHtmlDict], int]]
         ] = {
-            'choices': {'value': [
-                {
-                    'content_id': 'ca_choices_0',
-                    'html': '<p>This is value1 for ItemSelection</p>'
-                },
-                {
-                    'content_id': 'ca_choices_1',
-                    'html': '<p>This is value2 for ItemSelection</p>'
-                },
-                {
-                    'content_id': 'ca_choices_2',
-                    'html': '<p>This is value3 for ItemSelection</p>'
-                }
-            ]},
+            'choices': {'value': choices_subtitled_dict},
             'minAllowableSelectionCount': {'value': 1},
             'maxAllowableSelectionCount': {'value': 2}
         }
+        choices_subtitled_dict = [
+          {
+              'content_id': 'ca_choices_0',
+              'html': '<p>This is value1 for DragAndDropSortInput</p>'
+          },
+          {
+              'content_id': 'ca_choices_1',
+              'html': '<p>This is value2 for DragAndDropSortInput</p>'
+          }
+        ]
         customization_args_dict4: Dict[
-            str, Dict[str, Union[List[Dict[str, str]], bool]]
+            str, Dict[str, Union[List[state_domain.SubtitledHtmlDict], bool]]
```

Comment:
> Is there some reason you're splitting choices_subtitled_dict out? 

yes, because MyPy considers List of SubtitledHtmlDicts as a list of Dict[str, str] by itself. for eg:

```python
[
            {
                'content_id': 'ca_choices_0',
                'html': '<p>This is value1 for MultipleChoice</p>'
            },
            {
                'content_id': 'ca_choices_1',
                'html': '<p>This is value2 for MultipleChoice</p>'
            }
]
```

Here every element of List is Dict[str, str] if we just depends on MyPy, so to fix this behavior I just taken out these lists and annotate them explicitly.

Reply:
> Oh ok. Just to confirm, you're saying that if you leave them inline, MyPy will throw an error (even though you've annotated the "outer object" completely)?

yes